enhancements/single-node/single-node-openshift-with-workers.md

```diff
@@ -0,0 +1,483 @@
+
+- `SNO` = Single control-plane node cluster
+
+|Scenario                                |Platform|D1W|D2W|CIT|PIT|CTP|PTP|Is a load balancer currently needed?             |Will a load balancer be needed after proposal?|Current Support                     |
+|----------------------------------------|--------|---|---|---|---|---|---|-------------------------------------------------|----------------------------------------------|------------------------------------|
+|None platform SNO no workers            |None    |0  |0  |SR |SR |W  |M  |No                                               |No                                            |Works, supported, tested            |
+|Cloud platform SNO no workers           |Cloud   |0  |0  |SR |SR |W  |M  |No                                               |No                                            |Works, unsupported, tested          |
+|None platform SNO day1 single worker    |None    |1  |1  |-  |SR |-  |M  |Yes, ingress floats                              |No, ingress pinned to master pool             |Impossible to install               |
+|Cloud platform SNO day1 single worker   |Cloud   |1  |1  |SR |HA |W  |W  |Yes - provided, ingress floats                   |Yes - provided, there are two ingress replicas|Works, unsupported, untested        |
+|None platform SNO day1 multiple workers |None    |2+ |2+ |-  |SR |-  |M  |Yes, ingress is on two different nodes           |No, one ingress and is pinned to master pool  |Impossible to install               |
+|Cloud platform SNO day1 multiple workers|Cloud   |2+ |2+ |HA |HA |W  |W  |Yes - provided, ingress is on two different nodes|Yes - provided, there are two ingress replicas|Works, unsupported, untested        |
+|None platform SNO only day2 workers     |None    |0  |1+ |SR |SR |W  |M  |Yes, ingress floats                              |No, ingress pinned to master pool             |Ingress issue, unsupported, untested|
+|Cloud platform SNO only day2 workers    |Cloud   |0  |1+ |SR |SR |W  |M  |Yes - provided, ingress floats                   |Yes - provided, ingress floats                |Works, unsupported, untested        |
+
+Some notes about this table and the decisions made in it:
+
+The table assumes that the "Infrastrcture Topology" value is read only. That
+means an optimal value for this parameter must be determined during
+installation. You can see the consequences of that, for example, in the last
+row of the table - even though we add workers in day 2 to the cluster, the
+"Infrastrcture Topology" stays "SingleReplica" even though that cluster could
+benefit from the additional worker nodes for the purposes of highly-available
+infrastructure (as it's running in the cloud, so it has a load-balancer). In
+the future it may be worth revisiting making this value configurable.
+
+The "None platform SNO day1 single/multiple worker" scenario rows aren't really
+possible today as there is no real way to install this kind of cluster. That's
```

Comment:
I am missing what the limitation is that makes a SNO cluster with day-1 workers an impossibility for the "none" platform. I see mention above about the worker ignition file not being generated when doing a bootstrap-in-place installation. Is that what the "impossibility" referes to here? If so, then that assume that (1) the SNO installation is a bootstrap-in-place installation and (2) that the user does not know how to run `openshift-install create ignition-configs` to generate the worker ignition.

Reply:
Yeah at first the table specified the topology that such clusters would have according to the current installer logic for the number of workers, but I was convinced that this is not even possible, so I left them blank. 

I wasn't aware the installer is able to:

- Do single node "none" platform without bootstrap-in-place (using a bootstrap node). It's not something I ever tried myself / it's not something we're testing periodically so I guess I kinda just assumed it might not work.

- Didn't realize generating the worker configs in bootstrap-in-place is a simple as running this command... but when we're doing bootstrap-in-place how/when would that worker join? That that work in any way? Also, I guess the definition of "day-1" workers becomes a bit blurry. Does day-1 workers simply mean the number of workers specified in the installer-config manifest? I guess that definition could make sense. Would the installer even accept doing bip with workers being a non-0 value today?

The first bullet is convincing enough for me to fix the table to show the current topology rather than `-` though, so I'll do that.

In any case, the important bit in this part of the enhancement was to explicitly discuss / specify what the values *should* be today, rather than what it is now.